worker/firewaller/firewaller.go

```diff
@@ -333,6 +335,13 @@ func (fw *Firewaller) publishNetworkChanged(change *remoteRelationNetworkChange)
 		logger.Debugf("relation id not found publishing %+v", event)
 		return nil
 	}
+
+	// If the requested ingress is not permitted on the offering side,
+	// mark the relation as in error. It's not an error that requires a
+	// worker restart though.
+	if params.IsCodeForbidden(err) {
+		return fw.firewallerApi.SetRelationStatus(relData.tag.Id(), relation.Error, err.Error())
```

Comment:
it makes me a little uneasy having the firewaller set status on the relation, but I'm not sure what the alternative is. what else sets the status? how do they coordinate?

Reply:
Ostensibly, the other way a relation status can be set is if the user suspends the relation, but in that case, this firewall logic wouldn't be applied. Ingress is only requested when the relation is joined (again). So it hangs together as things stand. Juju sets status to joined, error, or broken. A user can set to suspended (but only if current status is joined, although that's not checked yet). There's a few loose ends to tidy up once we get feedback on this MVP.